web/client/themes/default/less/toc.less

```
@@ -88,6 +88,13 @@
             .background-color-var(@theme-vars[selected-bg]);
             .outline-color-var(@theme-vars[focus-color]);
         }
+        .wms-json-legend-rule:hover {
+            .color-var(@theme-vars[primary]);
+        }
+        .wms-json-legend-rule.active {
+            .background-color-var(@theme-vars[primary]);
+            .color-var(@theme-vars[primary-contrast]);
+        }
```

we could use the selected colors introduced in the TOC updates

```suggestion
        .wms-json-legend-rule:hover {
            .background-color-var(@theme-vars[selected-hover-bg]);
        }
        .wms-json-legend-rule.active {
            .background-color-var(@theme-vars[selected-bg]);
        }
        .wms-json-legend-rule.active {
            .background-color-var(@theme-vars[primary]);
            .color-var(@theme-vars[primary-contrast]);
        }
```